src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -428,3 +428,23 @@
    (fn [{:keys [locals]}]
      (.removeEventListener js/window "mouseup" (:on-mouse-up @locals))
      (.removeEventListener js/window "mousemove" (:on-mouse-move @locals)))})
+
+
+(react/defc TextFilter
+  {:render
+   (fn [{:keys [props this]}]
+     [:div {:style {:display "inline-flex"}}
+      (style/create-search-field
+        {:ref "filter-field" :autoSave "true" :results 5 :autofocus "true"
+         :placeholder "Filter" :defaultValue (:initial-text props)
+         :style {:borderRadius "3px 0 0 3px" :marginBottom 0}
+         :onKeyDown (common/create-key-handler [:enter] #(react/call :apply-filter this))})
```

Comment:
I'd love to see the caller be able to pass properties into the search field - for instance, the library page should pass "Search" as the placeholder text instead of "Filter". Look at cljs `merge` for this; there are instances of this in our code.